### PR TITLE
Retrieve localId via webdav

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Needed for https://github.com/nextcloud/android/pull/8643
Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: src/main/java/com/owncloud/android/lib/resources/activities/GetActivitiesRemoteOperation.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ public RemoteOperationResult run(NextcloudClient client) {
         String url = client.getBaseUri() + OCS_ROUTE_V12_AND_UP;
 
         // add filter for fileId, if available
-        if (!fileId.isEmpty()) {
+        if (fileId > 0) {
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/activities/GetActivitiesRemoteOperation.java:105]: "unitialized" value is actually ```-1``` so is 0 considered "empty" and thus the ```>0``` ?

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/lib/resources/activities/GetActivitiesRemoteOperation.java:105]: True, but fileId can never be 0 on server.
-1 is common for uninitialized.
But you are right that this is a bit strange.

**Review by AndyScherzinger — APPROVED** [review]

**Review by AndyScherzinger — APPROVED** [review]

**Review by AndyScherzinger — APPROVED** [review]

**Review by AlvaroBrey — DISMISSED** [review]

**Comment by tobiasKaminsky** [comment]
@AlvaroBrey 
```
java.lang.AssertionError: expected: java.lang.Long<69> but was: java.lang.String<69>
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failNotEquals(Assert.java:835)
at org.junit.Assert.assertEquals(Assert.java:120)
at org.junit.Assert.assertEquals(Assert.java:146)
at com.nextcloud.android.lib.resources.search.UnifiedSearchRemoteOperationIT.filesSearch(UnifiedSearchRemoteOperationIT.kt:78)
```
Attributes is Map<String, String>. 
Thus I dare not to touch it.

I now fixed the test, by converting long to String.

**Review by AlvaroBrey — APPROVED** [review]

**Review by AlvaroBrey — DISMISSED** [review]

**Review by AlvaroBrey — APPROVED** [review]

**Review by AlvaroBrey — DISMISSED** [review]

**Review by AlvaroBrey — APPROVED** [review]

**Pull request merged**